### PR TITLE
feat!: better integration of vscode multiple cursors

### PR DESCRIPTION
**Pull request by xiyaowong**
TODO
#97

### CODE REVIEW

File: runtime/lua/vscode-neovim/multi-cursors.lua [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+---@class Cursor
+---@field range LspRange
+---@field extmarks number[] cursor hilight and range(selection) highlight
+
+local CREATE_CURSOR_DEFER_TIME = 10
+local START_DEFER_TIME = 40
+local NOTIFY_DEFER_TIME = 20
+
+local ns = api.nvim_create_namespace("--- vscode multiple cursors ---")
```
Comment (justinmk) [inline runtime/lua/vscode-neovim/multi-cursors.lua:20]: nit: strange to use punctuation in a namespace name.
```suggestion
local ns = api.nvim_create_namespace("vscode.multicursor")
```

**Comment by xiyaowong** [comment]
Ready for review, but not ready for merge

**Comment by theol0403** [comment]
Btw @xiyaowong if you want to set up GitHub sponsors I'd be happy to add you to the list for the project.

**Comment by xiyaowong** [comment]
@theol0403 Thank you! Not for now, after all, I have only recently started contributing to the project.

**Comment by xiyaowong** [comment]
Lower priority of this PR, convert to draft for now

**Comment by theol0403** [comment]
@xiyaowong btw I have one week right now with no classes so I can be more involved this week and maybe even make PRs. I agree this feature is lower priority, but in general, if you want to work with me on PRs now is a good time.

**Comment by xiyaowong** [comment]
Great!

**Comment by xiyaowong** [comment]
@theol0403 This PR currently has the available functions, and you can test it.

https://github.com/vscode-neovim/vscode-neovim/assets/47070852/af31a3a5-dbef-4542-8955-4f45b58ba0d9

**Comment by xiyaowong** [comment]
By using flash.nvim, it is convenient to create cursors. This should be very practical.

cursors

**Comment by xiyaowong** [comment]
It's best as a built-in plugin. The current built-in multi-cursors is too rudimentary. And this PR does not change the current usage. However, a separate plugin is indeed more convenient to update.

**Comment by theol0403** [comment]
This PR is entirely contained within neovim, so no disadvantage to having it external. Creates maintanance burden for extension. 

Also, this integrates with external plugins (like flash). Ideally, VSCode neovim has no notion of the external ecosystem, it should be pure and vanilla. 

This seems like a perfect usecase of an extension making use of the new API.

**Comment by xiyaowong** [comment]
Should it be an "official" repo or my personal repo? Considering that this can be used as a use case of the new api and solve the existing issue, I prefer to be an "official" plugin. Users can improve this plugin together or create a better plugin.

**Comment by theol0403** [comment]
> Should it be an "official" repo or my personal repo? Considering that this can be used as a use case of the new api and solve the existing issue, I prefer to be an "official" plugin. Users can improve this plugin together or create a better plugin.

Official is good. It can be under vscode-neovim.

**Pull request closed without merging**